develop-csharp-plugins.md

```diff
@@ -1,7 +1,71 @@
-## How to develop C# plugins
+# DotNet plugins Development Guide
 
-We are currently writing up this section, will be updated soon.
+Flow is written in C#, so plugins written in dotnet platform can directly communicate with Flow without extra protocols.
 
-For now you can use the [dotnet template](https://github.com/Flow-Launcher/dotnet-template) to generate your plugin 
+## Initialization
+
+We recommand you use the [dotnet template](https://github.com/Flow-Launcher/dotnet-template) to generate a plugin template first.
+
+In order to be recongized as a Flow DotNet plugin, the directory need to have at least two files
+1. [`plugin.json`](plugin.json.md)
+2. A Dotnet class that implement **[IPlugin](API-Reference/Flow.Launcher.Plugin/iplugin.md)** or **[IAsyncPlugin](API-Reference/Flow.Launcher.Plugin/iasyncplugin.md)** (remember to refrence [Flow.Launcher.Plugin](https://www.nuget.org/packages/Flow.Launcher.Plugin/) by Nuget). The plugin template will add the reference and create  a`Main.cs` that implement `IPlugin`.
```

Comment:
change "implement" to "implements" (twice)

also fix the spacing for "create  a`Main.cs`"